lib/SILOptimizer/Mandatory/DiagnoseUnreachable.cpp

```diff
@@ -867,6 +868,86 @@ static bool simplifyBlocksWithCallsToNoReturn(SILBasicBlock &BB,
   return true;
 }
 
+/// Replaces  `switch_enum` / `switch_enum_addr` instructions that have cases
+/// matching unavailable enum elements with new instructions that have those
+/// cases removed since unavailable enum elements cannot be instantiated at
+/// run time. A default case that traps is also added if there was no default case
```

Comment:
As we talked about, this needs an explanation on why if there is already a default case, you don't map the unavailable cases to fatal error. Adding to this comment the explanation that you gave me would go a long way towards making this clearer.